--- setup-env.org
+++ setup-env
@@ -95,9 +95,9 @@
     echo "Usage: . $PROGNAME -m <machine>"
 
     echo -e "\n    Supported machines:
-    `find ${ROOTDIR}/sources/meta-{fsl,freescale,qoriq-demos}*/conf/machine \
+    `find ${ROOTDIR}/sources/meta-{fsl,freescale,qoriq-demos,scalys}*/conf/machine \
         -name *.conf 2>/dev/null \
-        |egrep "/(mpc|b4|bsc|c29|p1|p2|p3|p4|p5|t1|t2|t4|ls|lx).*conf$" |sort \
+        |egrep "/(mpc|b4|bsc|c29|p1|p2|p3|p4|p5|t1|t2|t4|ls|lx|qls).*conf$" |sort \
         |xargs echo |sed -e 's,[^ ]*/,,g;s,.conf,,g'`
 "
 
@@ -154,11 +154,11 @@
 # Check the machine type specified
 valid_machine=false
 if [ -n "${MACHINE}" ];then
-    valid_num=`find ${ROOTDIR}/sources/meta-{fsl,freescale,qoriq-demos}*/conf/machine \
+    valid_num=`find ${ROOTDIR}/sources/meta-{fsl,freescale,qoriq-demos,scalys}*/conf/machine \
         -name ${MACHINE}.conf 2>/dev/null |wc -l`
     if [ "1" -lt "$valid_num" ];then
         echo "ERROR: possible error may occur due to duplicate ${MACHINE}.conf exist:"
-        find ${ROOTDIR}/sources/meta-{fsl,freescale,qoriq-demos}*/conf/machine \
+        find ${ROOTDIR}/sources/meta-{fsl,freescale,qoriq-demos,scalys}*/conf/machine \
             -name ${MACHINE}.conf 2>/dev/null
         echo "Please remove the useless layer under ${ROOTDIR}/sources"
         clean_up && return
@@ -187,6 +187,7 @@
     meta-freescale \
     meta-freescale-distro \
     meta-qoriq-demos \
+    meta-scalys \
 "
 EULA_FILE="$FSLROOTDIR/EULA"
 
